File: 백준/Bronze/9094. 수학적 호기심/수학적 호기심.c
#include <stdio.h>
int main(void) {
    int t, n, m, cnt;
    scanf("%d", &t);
    for (int i = 0; i < t; i++) {
        scanf("%d %d", &n, &m);
        cnt = 0;
        for (int j = 1; j < n; j++) {
            for (int k = j + 1; k < n; k++) {
                if ((j * j + k * k + m) % (j * k) == 0)
                    cnt += 1;
            }
        }
        printf("%d\n", cnt);
    }
    return 0;
}

File: 백준/Bronze/9094. 수학적 호기심/수학적 호기심.py
import sys
input=lambda:sys.stdin.readline().rstrip()
for _ in range(int(input())):
    n,m=map(int,input().split())
    cnt=0
    for i in range(1,n):
        for j in range(i+1,n):
            if (i**2+j**2+m)%(i*j)==0:
                cnt+=1
    print(cnt)